tools/release/__init__.py

```diff
@@ -173,12 +168,17 @@ def isolated_temporary_checkout(
         git_checkout_dir = tmp_dir / f"pip-build-{target_ref}"
         nox_session.run(
             # fmt: off
-            "git", "clone",
-            "--depth", "1",
-            "--config", "core.autocrlf=false",
-            "--branch", str(target_ref),
+            "git",
+            "clone",
+            "--depth",
+            "1",
+            "--config",
+            "core.autocrlf=false",
+            "--branch",
+            str(target_ref),
             "--",
-            ".", str(git_checkout_dir),
+            ".",
+            str(git_checkout_dir),
             # fmt: on
```

Comment:
Looks like ruff is flat-out ignoring `#fmt: off` here. That's a ruff bug (or at best, an annoying decision to not respect the standard convention) and should be reported to ruff. IMO this is a blocker to us adopting ruff.

Reply:
I think this is Ruff needing the comments at statement level.

See https://docs.astral.sh/ruff/formatter/#format-suppression which says instead of:

```python
[
    # fmt: off
    '1',
    # fmt: on
    '2',
]
```
Do:

```python
# fmt: off
[
    '1',
    '2',
]
# fmt: on
```

Reply:
Understood. I still think it's a problem with ruff, though. At the absolute minimum it should warn if it's ignoring formatting directives. (Specifically if it's ignoring directives that other formatters like black would respect).